### PR TITLE
doc/dev: doc/dev/osd_internals capitalization, formatting, clarity

### PR DESCRIPTION
**Pull request by anthonyeleven**
doc/dev: spruce up several doc/dev/osd_internals files for caps, markup, formatting, clarity 

Signed-off-by: Anthony D'Atri <anthony.datri@gmail.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by anthonyeleven** [comment]
jenkins test docs

**Comment by anthonyeleven** [comment]
jenkins test docs

**Comment by anthonyeleven** [comment]
@zdover23 This one is ready.

**Comment by zdover23** [comment]
Change A to B:

A:

Ceph Placement Groups (PGs) maintain a log of write transations to faciliate
speedy recovery of data.  When recovering, Instead of scanning all RADOS
objects to find what is missing or outdated within each OSD we can determine
this information each PG's log.

B: Ceph Placement Groups (PGs) maintain a log of write transactions to
facilitate speedy recovery of data. During recovery, each of these PG logs
is used to determine which content in each OSD is missing or outdated.
This obviates the need to scan all RADOS objects.

Comments
---------------
1. "transations" is missing the "c" that would make it "transactions".
2. I've rewritten the rest of the sentence so that the sentence's head (the PG logs, or, arguably, their function in data recovery) is closer to the front of the sentence and likely foregrounded in the reader's mind.

**Comment by zdover23** [comment]
Change A to B:

A:

The threashold for performing asynchronous recovery instead of synchronous
recovery is not a clear-cut. There are a few criteria which
need to be met for asynchronous recovery:

B: The threshold that triggers asynchronous recovery is not clear-cut. 
The following criteria must be met to trigger asynchronous recovery:

**Comment by zdover23** [comment]
Change A to B:

A:
When a new OSD joins a cluster all PGs with it in their acting sets must
eventually backfill.  If all of these backfills happen simultaneously
they will present excessive load on the OSD:  the "thundering herd"
effect. 

B: When a new OSD joins a cluster, each PG that has it in its acting set
must eventually backfill. If all of these backfills happen simultaneously
they will put excessive load on the OSD: this is the "thundering herd"
effect.

**Comment by zdover23** [comment]
Change A to B:

A:
since ``osd.1`` is the only one which recorded ``info.les=477`` while ``osd.4``,``osd.0``
which were the acting set in that interval did not (``osd.4`` restarted and ``osd.0``
did not get the message in time) the PG is marked incomplete when
either ``osd.4`` or ``osd.0`` would have been valid choices.  To avoid this, we do not

B:
since ``osd.1`` is the only one which recorded ``info.les=477``, while ``osd.4``,``osd.0``
(which were the acting set in that interval) did not (``osd.4`` restarted and ``osd.0``
did not get the message in time), the PG is marked incomplete when
either ``osd.4`` or ``osd.0`` would have been valid choices.  To avoid this, we do not

Comment
-------------
This change is pretty finicky, but I think that the addition of the commas and the parentheses clean up the sentence and make its syntax a little clearer, reducing cognitive load on the reader.

**Comment by zdover23** [comment]
> @zdover23 This one is ready.

I've looked through this and made some notes below in a few places to improve the syntax of some of the sentences and to reduce the cognitive load on the reader. I'm not technically savvy enough to verify the updated technical content in this edit, and so my esteem for your work carries me immediately forth to assume that the technical updates here are all correct. This is a good, solid contribution, and is also a foretaste of the upcoming tedious, line-by-line reworking of the Developer Guide.

As soon as these alterations are made, I'll approve this one.

**Comment by zdover23** [comment]
> > @zdover23 This one is ready.
> 
> I've looked through this and made some notes below in a few places to improve the syntax of some of the sentences and to reduce the cognitive load on the reader. I'm not technically savvy enough to verify the updated technical content in this edit, and so my esteem for your work carries me immediately forth to assume that the technical updates here are all correct. This is a good, solid contribution, and is also a foretaste of the upcoming tedious, line-by-line reworking of the Developer Guide.
> 
> As soon as these alterations are made, I'll approve this one.

Well, I wrote "below", thinking that github would put that comment above the comments I was referring to, but I clearly should have written "above". So: as above, so... above.

**Review by zdover23 — CHANGES_REQUESTED** [review]
See the comments in the "Conversation" page. These constitute my requested changes. When these changes are made (or convincing cases for not making them are articulated), I'll approve this one.

Excelsior and qapla!

File: doc/dev/deduplication.rst [hunk 1]
```diff
@@ -89,12 +89,12 @@ scheme between replication and erasure coding depending on
...
 
-Regarding how to use, please see osd_internals/manifest.rst
+Regarding how to use, please see ``osd_internals/manifest.rst``
 
 Usage Patterns
 ==============
 
-The different ceph interface layers present potentially different oportunities
+The different Ceph interface layers present potentially different oportunities
 and costs for deduplication and tiering in general.
```
Comment (anthonyeleven) [inline doc/dev/deduplication.rst:98]: test

File: doc/dev/deduplication.rst [hunk 2]
```diff
@@ -131,21 +131,21 @@ support needs robust support for snapshots.
...
 
 Status and Future Work
 ======================
 
 At the moment, there exists some preliminary support for manifest
-objects within the osd as well as a dedup tool.
+objects within the OSD as well as a dedup tool.
 
 RadosGW data warehouse workloads probably represent the largest
 opportunity for this feature, so the first priority is probably to add
```
Comment (anthonyeleven) [inline doc/dev/deduplication.rst:144]: s/feature/creature/

**Comment by anthonyeleven** [comment]
@zdover23 Suggestions all seem reasonable to me and I've pushed an amended commit.

Suggest that going forward a more effective feedback strategy is in-line in each file under the "N files changed" tab, hovering between the line numbers and the text brings up a blue `+` which allows `Add single comment`.  This way you don't have to quote the existing text, as the comment's location implies that, and also the submitter don't have to dig through the files to figure out which one each feedback refers to.  This also facilitates independent discussion as well as resolution of each discussion thread as it's sorted out.

**Comment by zdover23** [comment]
> @zdover23 Suggestions all seem reasonable to me and I've pushed an amended commit.
> 
> Suggest that going forward a more effective feedback strategy is in-line in each file under the "N files changed" tab, hovering between the line numbers and the text brings up a blue `+` which allows `Add single comment`. This way you don't have to quote the existing text, as the comment's location implies that, and also the submitter don't have to dig through the files to figure out which one each feedback refers to. This also facilitates independent discussion as well as resolution of each discussion thread as it's sorted out.

Roger wilco.

**Review by zdover23 — APPROVED** [review]

**Pull request merged**